src/interfaces/node.h

```diff
@@ -53,6 +53,7 @@ class EVO
 public:
     virtual ~EVO() {}
     virtual std::pair<CDeterministicMNList, const CBlockIndex*> getListAtChainTip() = 0;
+    virtual void setContext(NodeContext* context) {}
```

Comment:
IMO we should not pass here the whole `NodeContext` which has way too many things inside.
Why don't pass directly deterministicMNManager if that's needed?

Reply:
The argument list for `setContext` was intentionally mirroring the parent `NodeContext` class that will be calling it, we are implementing these as part of an interface, I suggest that the argument list should remain the same for all interfaces as long as it carries the same name.

An additional reason why we're passing the entire `NodeContext` is, imo, further deglobalization efforts will find that certain operations are dependent on multiple Dash components (as becomes apparent with [`getObjAbsYesCount`](https://github.com/dashpay/dash/blob/26a11d75643f2401cab4c3f82bd237157a9f30d5/src/node/interfaces.cpp#L115-L121)).